tests/test_engine/test_model_client.py

```
@@ -96,6 +96,28 @@ def test_parse_response_rejects_empty_choices(self):
+        resp.status = 200
+        resp.raise_for_status = MagicMock()
+        resp.json = AsyncMock(return_value=None)
+
+        post_cm = MagicMock()
+        post_cm.__aenter__ = AsyncMock(return_value=resp)
+        post_cm.__aexit__ = AsyncMock(return_value=False)
+        fake_client = MagicMock()
+        fake_client.post = MagicMock(return_value=post_cm)
+
+        with patch.object(c, "_get_client", return_value=fake_client):
+            with pytest.raises(InfraError, match="non-object JSON"):
```

Ruff (0.15.21)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)